src/auto/subprocess_launcher.rs

```diff
@@ -0,0 +1,220 @@
+// This file was generated by gir (https://github.com/gtk-rs/gir)
+// from gir-files (https://github.com/gtk-rs/gir-files)
+// DO NOT EDIT
+
+#[cfg(any(feature = "v2_40", feature = "dox"))]
+use Error;
+#[cfg(any(feature = "v2_40", feature = "dox"))]
+use Subprocess;
+#[cfg(any(feature = "v2_40", feature = "dox"))]
+use SubprocessFlags;
+use ffi;
+use glib;
+use glib::object::Downcast;
+use glib::object::IsA;
+use glib::signal::SignalHandlerId;
+use glib::signal::connect;
+use glib::translate::*;
+use glib_ffi;
+use gobject_ffi;
+use std;
+use std::boxed::Box as Box_;
+use std::mem;
+use std::mem::transmute;
+use std::ptr;
+
+glib_wrapper! {
+    pub struct SubprocessLauncher(Object<ffi::GSubprocessLauncher>);
+
+    match fn {
+        get_type => || ffi::g_subprocess_launcher_get_type(),
+    }
+}
+
+impl SubprocessLauncher {
+    #[cfg(any(feature = "v2_40", feature = "dox"))]
+    pub fn new(flags: SubprocessFlags) -> SubprocessLauncher {
+        unsafe {
+            from_glib_full(ffi::g_subprocess_launcher_new(flags.to_glib()))
+        }
+    }
+}
+
+pub trait SubprocessLauncherExt {
+    #[cfg(any(feature = "v2_40", feature = "dox"))]
+    fn getenv<P: AsRef<std::path::Path>>(&self, variable: P) -> Option<std::path::PathBuf>;
+
+    //#[cfg(any(feature = "v2_40", feature = "dox"))]
+    //fn set_child_setup(&self, child_setup: /*Unknown conversion*//*Unimplemented*/SpawnChildSetupFunc, destroy_notify: /*Unknown conversion*//*Unimplemented*/DestroyNotify);
+
+    #[cfg(any(feature = "v2_40", feature = "dox"))]
+    fn set_cwd<P: AsRef<std::path::Path>>(&self, cwd: P);
+
+    #[cfg(any(feature = "v2_40", feature = "dox"))]
+    fn set_environ(&self, env: &[&std::path::Path]);
+
+    #[cfg(any(feature = "v2_40", feature = "dox"))]
+    fn set_flags(&self, flags: SubprocessFlags);
+
+    #[cfg(any(feature = "v2_40", feature = "dox"))]
+    fn set_stderr_file_path<P: AsRef<std::path::Path>>(&self, path: P);
+
+    #[cfg(any(feature = "v2_40", feature = "dox"))]
+    fn set_stdin_file_path(&self, path: &str);
+
+    #[cfg(any(feature = "v2_40", feature = "dox"))]
+    fn set_stdout_file_path<P: AsRef<std::path::Path>>(&self, path: P);
+
+    #[cfg(any(feature = "v2_40", feature = "dox"))]
+    fn setenv<P: AsRef<std::path::Path>, Q: AsRef<std::path::Path>>(&self, variable: P, value: Q, overwrite: bool);
```

Comment:
`OsString` as above (or rather they should all be `&OsStr`, there's a gir parameter for this)